src/bin/driver.rs

```diff
@@ -600,20 +616,21 @@ mod app {
                 c.local.internal_adc.read_output_voltage(Channel::HighPower),
             ],
         );
-        let interlock_asserted = c
-            .shared
-            .laser_interlock_pin
-            .lock(|pin| pin.get_state() == PinState::High);
+        let interlock_asserted =
+            c.shared.laser_interlock.lock(|ilock| ilock.state.is_none());
         for (i, (interlock_current, read)) in interlock_current
             .iter()
             .zip((current_reads).iter())
             .enumerate()
         {
             if (read > interlock_current) & interlock_asserted {
-                c.shared.laser_interlock_pin.lock(|pin| pin.set_low());
+                let ch = Channel::try_from(i).unwrap();
+                c.shared.laser_interlock.lock(|ilock| {
+                    ilock.set(Some(LaserInterlockTripped::Overcurrent(ch)))
```

Comment:
`interlock_asserted`  becomes inconsistent with the pin and and `ilock.state` becomes inconsistent with the state that caused the interlock to fire!

Reply:
Ok, but does that matter if `interlock_asserted` is dropped anyways at the end of the task? If we have an overvoltage condition as well that would trp the interlock a second time, but that isn't a problem either right?